sdk/dotnet/Pulumi/Deployment/GrpcEngine.cs

```diff
@@ -19,13 +19,13 @@ public GrpcEngine(string engine)
             this._engine = new Engine.EngineClient(new Channel(engine, ChannelCredentials.Insecure, grpcChannelOptions));
         }
         
-        public async Task LogAsync(LogRequest request)
-            => await this._engine.LogAsync(request);
+        public Task LogAsync(LogRequest request)
+            => this._engine.LogAsync(request).ResponseAsync;
```

Comment:
What is `ResponseAsync`?

Reply:
This has the same warning AsyncFixer01 since the await is effectively redundant.

The return type is `AsyncUnaryCall<TResponse>` which is awaitable via its `GetAwaiter` method:
```c#
public TaskAwaiter<TResponse> GetAwaiter()
{
    return responseAsync.GetAwaiter();
}
```
The `responseAsync` field holds the `Task` that can be awaited:
```c#
readonly Task<TResponse> responseAsync;
```
And this field is otherwise accessible via the `ResponseAsync` property:
```c#
public Task<TResponse> ResponseAsync
{
    get
    {
        return this.responseAsync;
    }
}
```
So to remove the redundant await I return the underlying task via this property.